### ISSUE TITLE
Issue with password after reset

### ISSUE THREAD

**Issue by dmaechler** [thecodingmachine/security.forgot-your-password#4]
When a password is reset, it is saved in clear in database via the method setPasswordAndDiscardToken() of the SecurityUserDao.php file line 158.

We should have the possibility to at least encode it into sha1 or md5...

**Comment by moufmouf** [comment]
Is this the right package to post this issue?

If I'm right, `SecurityUserDao.php` is not part of this package.

And by the way, we **never** encode a password in sha1 or md5. We use `password_hash` since PHP 5.5.

**Comment by dmaechler** [comment]
Yes maybe. I didn't know if it was the role of the forgot password service to hash the password or to the security dao ... We should add this line 
$password = password_hash($password, PASSWORD_BCRYPT);
to the security dao setPasswordAndDiscardToken() then. Should I make a PR ?

**Comment by moufmouf** [comment]
No.

The code for setPasswordAndDiscardToken is: 

```php
    public function setPasswordAndDiscardToken(string $token, string $password)
    {
        $user = $this->getUserByToken($token);
        $user->setPassword($password);
        $user->setToken(null);
        $this->tdbmService->save($user);
    }
```

The documentation states that the `UserBean` MUST use `UserTrait`.

UserTrait is in charge of replacing setPassword with a hashed version of setPassword.

So the problem stems from users not reading the bloody doc.

That being said, I guess it is foolish for me to believe that users will ever read the doc, right?

Maybe we should modify the trait to add another "hashAndSetPassword" method?

That way, if the method does not exist in the UserBean, it will trigger an error and the user will be forced to realize he forgot to add the trait?

If you write a PR? Write it this way.
